packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -85,8 +106,11 @@ export type StoredJobInformation = {
 
 export type CronjobControllerState = {
   jobs: Record<string, StoredJobInformation>;
+  events: Record<string, BackgroundEvent>;
 };
 
+const subscriptionMap = new WeakMap();
```

Comment:
I'm really not a fan of this, there has to be a better way to handle teardown of controllers 🫠 

I'll note again that the teardown might not be important regardless due to the client life-cycle. E.g. controllers aren't destroyed unless the extension is being shut down entirely anyway.

Reply:
Yeah I know, I mostly added it in because it seemed like we were already using destroy.

Reply:
Didn't the hack we had before here work?

Reply:
We just didn't unsubscribe i think

Reply:
I think we were? https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/cronjob/CronjobController.ts#L345-L368

Reply:
ohh I thought you meant the work I did in the `SnapInterfaceController`

Reply:
what was the hack before? I don't follow

Reply:
Oh the problem was existing before I removed those lines, I just removed the variable reassignments since I was already binding and storing in the map

Reply:
I don't follow this, we had it setup before where the destroy function was working fine and was tested without using a `Map` but instead using a different hack.

Reply:
Ahh that's my fault, I hadn't bound the new event handlers which was the cause of all of this. I thought it was breaking for some other reason, all good now!

Reply:
Yes, just pushed.